### ISSUE TITLE
gpt-pilot fails to continue on an existing project and lost everything

### ISSUE THREAD

**Issue by pezzos** [Pythagora-io/gpt-pilot#197]
When I exit a run with CTRL+C and want to come back later with:
`python main.py app_id=a1cd161d-bd6e-4040-976e-79ac8f6eef36 step=62`

gpt-pilot start from the begin, hence erase ALL the steps and the, fail with this error:
```
---------- GPT PILOT EXITING WITH ERROR ----------
Traceback (most recent call last):
  File "/Users/pezzos/Dev/gpt-pilot/pilot/main.py", line 53, in <module>
    project.start()
  File "/Users/pezzos/Dev/gpt-pilot/pilot/helpers/Project.py", line 141, in start
    self.developer.start_coding()
  File "/Users/pezzos/Dev/gpt-pilot/pilot/helpers/agents/Developer.py", line 41, in start_coding
    self.implement_task(i, dev_task)
  File "/Users/pezzos/Dev/gpt-pilot/pilot/helpers/agents/Developer.py", line 70, in implement_task
    return self.execute_task(convo_dev_task, task_steps, development_task=development_task, continue_development=True, is_root_task=True)
  File "/Users/pezzos/Dev/gpt-pilot/pilot/helpers/agents/Developer.py", line 272, in execute_task
    result = self.step_human_intervention(convo, step)
  File "/Users/pezzos/Dev/gpt-pilot/pilot/helpers/agents/Developer.py", line 146, in step_human_intervention
    return_value = self.debugger.debug(convo, user_input=response['user_input'], issue_description=step['human_intervention_description'])
  File "/Users/pezzos/Dev/gpt-pilot/pilot/helpers/Debugger.py", line 44, in debug
    debugging_plan = convo.send_message('dev_ops/debug.prompt',
  File "/Users/pezzos/Dev/gpt-pilot/pilot/helpers/AgentConvo.py", line 91, in send_message
    response = parse_agent_response(response, function_calls)
  File "/Users/pezzos/Dev/gpt-pilot/pilot/utils/function_calling.py", line 73, in parse_agent_response
    values = list(json.loads(text).values())
  File "/Users/pezzos/devtools/pyenv/versions/3.10.2/lib/python3.10/json/__init__.py", line 346, in loads
    return _default_decoder.decode(s)
  File "/Users/pezzos/devtools/pyenv/versions/3.10.2/lib/python3.10/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/Users/pezzos/devtools/pyenv/versions/3.10.2/lib/python3.10/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
--------------------------------------------------
```

It happens to me multiple times.

I have the last version as I checkout ever time I run this.

1 - Could you please have a look to fix this?
2 - Is there a way to get all the code I lost (and pay for)?

**Comment by barath-uni** [comment]
By looking at this error message, I don't find it helpful to debug further 

Some questions to help the devs actually debug this faster.

1. Did anything happen before this error got thrown? What stage was it in? (Environment installation, writing code)
2. Do you not have the code written so far in a folder within the project named `workspace`
3. What db did you configure at the start?

PS: I am not a dev with the project.

**Comment by barath-uni** [comment]
Debugging the code further, it looks like the GPT did not give any response for a prompt.

Could be because you hit the rate limit, or a temporary bug. It's difficult to say.
Go to `pilot/helpers/AgentConvo.py` and in `line 90` print the `response` variable.

**Comment by nalbion** [comment]
should be resolved by #211
